tests/tracing/test_dnacurvature.py

```diff
@@ -0,0 +1,941 @@
+                    0.01719142,
+                    0.01719142,
+                    0.05832825,
+                    0.55555556,
+                    0.05832825,
+                    0.01719142,
+                    0.01719142,
+                    0.05832825,
+                ]
+            ),
+        ),
+    ],
+)
+def test_curvature_ellipse(
+    points,
+    major,
+    minor,
+    displacement,
+    edge_order,
+    expected_first_derivative,
+    expected_second_derivative,
+    expected_local_curvature,
+) -> None:
+    ellipse_points = ellipse_coordinates(points, major, minor, displacement)
+    curvature = Curvature(molecule_coordinates=ellipse_points, circular=True)
+    curvature.calculate_derivatives(edge_order=edge_order)
+    curvature._calculate_local_curvature()
+    np.testing.assert_array_almost_equal(curvature.first_derivative, expected_first_derivative)
+    np.testing.assert_array_almost_equal(curvature.second_derivative, expected_second_derivative)
+    np.testing.assert_array_almost_equal(curvature.local_curvature, expected_local_curvature)
+
+
+@pytest.mark.parametrize(
+    "points, major, minor, displacement, shift, edge_order, expected_first_derivative, expected_second_derivative, expected_local_curvature",
+    [
+        (
+            10,
+            5.0,
+            1.0,
+            0.0,
+            3,  # Shift array by 3 points
+            1,
+            np.asarray(
+                [
+                    [-4.44089210e-16, 5.87785252e-01],
+                    [1.72745751e00, 4.75528258e-01],
+                    [2.79508497e00, 1.81635632e-01],
+                    [2.79508497e00, -1.81635632e-01],
+                    [1.72745751e00, -4.75528258e-01],
+                    [8.88178420e-16, -5.87785252e-01],
+                    [-1.72745751e00, -4.75528258e-01],
+                    [-2.79508497e00, -1.81635632e-01],
+                    [-2.79508497e00, 1.81635632e-01],
+                    [-1.72745751e00, 4.75528258e-01],
+                ]
+            ),
+            np.asarray(
+                [
+                    [1.72745751e00, -2.77555756e-17],
+                    [1.39754249e00, -2.03074810e-01],
+                    [5.33813729e-01, -3.28581945e-01],
+                    [-5.33813729e-01, -3.28581945e-01],
+                    [-1.39754249e00, -2.03074810e-01],
+                    [-1.72745751e00, -2.77555756e-17],
+                    [-1.39754249e00, 2.03074810e-01],
+                    [-5.33813729e-01, 3.28581945e-01],
+                    [5.33813729e-01, 3.28581945e-01],
+                    [1.39754249e00, 2.03074810e-01],
+                ]
+            ),
+            np.asarray(
+                [5.0, 0.1765308, 0.04620573, 0.04620573, 0.1765308, 5.0, 0.1765308, 0.04620573, 0.04620573, 0.1765308]
+            ),
+        ),
+        (
+            10,
+            5.0,
+            1.0,
+            0.0,
+            3,  # Shift array by 5 points
```

Comment:
Same issue as before; I'd have changed it myself but I am not sure if you'd want to change the code or the comment